fighthealthinsurance/form_utils.py

```diff
@@ -24,10 +24,13 @@ def magic_combined_form(
     forms_to_merge: list[forms.Form], existing_answers: dict[str, str]
 ) -> forms.Form:
     combined_form = forms.Form()
+    seen_fields = set()
+
     for f in forms_to_merge:
         for field_name, field in f.fields.items():
-            # Add new fields to the combined form
-            if field_name not in combined_form.fields:
+            # Only add field if we haven't seen it before
+            if field_name not in seen_fields:
+                seen_fields.add(field_name)
```

Comment:
This changes seems un-needed, we already check fields for presence -- was that not working?

Reply:
In the denial e2e test, duplicates were showing up in the question list and putting that dedup there pulled them out